Mathlib/SetTheory/Ordinal/FixedPointApproximants.lean

```diff
@@ -110,6 +110,41 @@ theorem lfpApprox_add_one (h : x ≤ f x) (a : Ordinal) :
     simp only [Set.mem_setOf_eq]
     use a
 
+theorem lfpApprox_mono_left : Monotone (lfpApprox : (α →o α) → _) := by
+  intro f g h x a
+  induction a using Ordinal.induction with
+  | h i ih =>
+    unfold lfpApprox
```

Comment:
Having to unfold a definition over and over is somewhat of an antipattern. The issue here is that since your definition of `lfpApprox` is recursive, it's not unfolded by default, causing it not to be definitionally equal. A simple fix is to introduce a lemma like
```
theorem lfpApprox_def (a : Ordinal) :
    lfpApprox f x a = sSup ({ f (lfpApprox f x b) | (b : Ordinal) (_ : b < a) } ∪ {x}) := by
  rw [lfpApprox]
```
and rewrite using that lemma (which notably, is now searchable) instead of using `unfold`.

Reply:
I added that theorem and refactored it in the previous existing theorems. Is this fine or should the second part be a separate PR?